src/ruby/puppetserver-lib/puppet/server/execution.rb

```diff
@@ -7,31 +7,41 @@ class Puppet::Server::Execution
   def self.initialize_execution_stub
     Puppet::Util::ExecutionStub.set do |command, options, stdin, stdout, stderr|
       if command.is_a?(Array)
+        orig_command_str = command.join(" ")
         binary = command.first
         args = command[1..-1]
       else
+        orig_command_str = command
         binary = command
         args = nil
       end
 
-      # TODO - options is currently ignored - https://tickets.puppetlabs.com/browse/SERVER-74
+      # TODO - options relating to stdout/stderr are not yet handled, see SERVER-74 and
+      #  PUP-6640
 
       # We're going to handle STDIN/STDOUT/STDERR in java, so we don't need
       # them here.  However, Puppet::Util::Execution.execute doesn't close them
       # for us, so we have to do that now.
       [stdin, stdout, stderr].each { |io| io.close rescue nil }
 
-      execute(binary, args)
+      execute(orig_command_str, binary, options, args)
     end
   end
 
-  def self.execute(command, args = nil)
+  private
+  def self.execute(orig_command_str, command, options, args)
```

Comment:
I'm not clear on why having the caller pass in the `orig_command_str` is better than having this method compute this value. It seems like this method has all the information to create that value, that `orig_command_str` is only used in logging, and creating the log message from the other args would be within this method's responsibilities.

I could see a case for passing this argument in to improve the method's testability, without which we would have to have tests understand the implementation of how the method computed the log message. However, since the consumer of the test is `initialize_execution_stub` and we aren't testing that method, our test helper has to know the implementation of how `initialize_execution_stub` will compute the `orig_command_str` value still (ie we're coupling the implementation at line 10 & 14 in this file with lines 6 through 10 in the helper and the regex in the assertion at line 61 in the execution_spec).